### PR TITLE
lint: Convert Python dead code linter test to Python

### PR DESCRIPTION
**Pull request by fjahr**
The new python version should produce the exact same output as the bash version but be easier to maintain.

### CODE REVIEW

File: test/lint/lint-python-dead-code.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+def main():
+    check_vulture_install()
+
+    files = check_output(FILES_ARGS).decode("utf-8").splitlines()
+    # --min-confidence 100 will only report code that is guaranteed to be unused within the analyzed files.
+    # Any value below 100 introduces the risk of false positives, which would create an unacceptable maintenance burden.
+    vulture_args = ['vulture', '--min-confidence=100'] + files
+
+    try:
+        check_output(vulture_args, stderr=STDOUT)
```
Comment (MarcoFalke) [inline test/lint/lint-python-dead-code.py:33]: Will this print the error? I'd guess no

**Review by maflcko — APPROVED** [review]

File: test/lint/lint-python-dead-code.py (hunk 1 above)
Comment (fjahr) [reply test/lint/lint-python-dead-code.py:33]: Hm, for me it does. It doesn't for you?

(faked to get an alert)
```
$ test/lint/lint-python-dead-code.py
/Users/FJ/projects/clones/bitcoin/test/lint/lint-python-dead-code.py:26: unsatisfiable 'if' condition (100% confidence)
Python dead code detection found some issues
```

Comment (MarcoFalke) [reply test/lint/lint-python-dead-code.py:33]: Thanks for testing and confirming. I just assumed that redirecting stderr to stdout and then capturing stdout, which is done by `check_output`, which discards the return value here, would also drop the stderr.

This is also what the documentation says:

https://docs.python.org/3/library/subprocess.html#subprocess.check_output

> To also capture standard error in the result, use `stderr=subprocess.STDOUT`


Not sure what to do now. Are you trying to trick me into installing `vulture`? :thinking:

Comment (MarcoFalke) [reply test/lint/lint-python-dead-code.py:33]: Oh, sorry. I missed the `e.output.decode` two lines below. :man_facepalming:

**Comment by MarcoFalke** [comment]
review ACK 076cd6835fd97a62bfd6912b80addfcb5342ea8e

The three (3) CI failures are unrelated and can be ignored (:smiling_face_with_tear:)

**Pull request merged**